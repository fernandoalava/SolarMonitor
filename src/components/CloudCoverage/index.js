import React from "react";
import PropTypes from "prop-types";
import * as moment from "moment";
import CircularProgress from "@material-ui/core/CircularProgress";
import {
  AreaChart,
  Area,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend
} from "recharts";

export const CloudCoverage = props => {
  const { activity } = props;
  const now = moment();
  const next24hours = moment().add(24, "hours");

  const data = props.activity
    .filter(item => moment(item.axes.time).isBetween(now, next24hours))
    .map(item => ({
      time: moment(item.axes.time).format("HH:mm"),
      sky_clarity: Math.floor(item.data.av_ttl_cld * 100)
    }));

  return activity.length >= 1 ? (
    <AreaChart
      width={500}
      height={200}
      data={data}
      margin={{
        top: 5,
        right: 30,
        left: 20,
        bottom: 5
      }}
    >
      <CartesianGrid strokeDasharray="3 3" />
      <XAxis dataKey="time" />
      <YAxis dataKey="sky_clarity" />
      <Tooltip />
      <Legend />
      <Area
        type="monotone"
        dataKey="sky_clarity"
        stroke="#8884d8"
        activeDot={{ r: 8 }}
        name="% Sky Clarity"
      />
    </AreaChart>
  ) : (
    <CircularProgress />
  );
};

CloudCoverage.propTypes = {
  activity: PropTypes.array.isRequired
};

CloudCoverage.defaultProps = {
  activity: []
};

export default CloudCoverage;
