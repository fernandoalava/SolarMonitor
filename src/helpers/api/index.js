import config from "../../config";
import axios from "axios";

export const fetchCloudData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_ttl_cld`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};

export const fetchSolarData = async (dataHandler, errorHanlder) => {
  try {
    let response = await axios.get(
      `${config.protocol}://${
        config.planetOSDomain
      }/v1/datasets/bom_access-g_global_40km/point?origin=dataset-details&apikey=${
        config.apiKey
      }&count=50&lat=49.6&lon=-50.5&var=av_swsfcdown`
    );
    dataHandler(response.data.entries);
  } catch (err) {
    errorHanlder(err);
  }
};
