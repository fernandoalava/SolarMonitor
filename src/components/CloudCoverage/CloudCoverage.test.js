import { shallow } from "enzyme";
import * as React from "react";
import CloudCoverage from ".";
import { cloudCoverageDataset } from "../../helpers/index";

jest.mock("moment", () => () => ({
  format: () => "2018–01–30T12:34:56+00:00",
  add: () => "2018–01–30T12:34:56+00:00",
  isBetween: () => "2018–01–30T12:34:56+00:00"
}));

describe("CloudCoverage", () => {
  it("Renders correctly without data", () => {
    const wrapper = shallow(<CloudCoverage activity={[]} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Renders correctly with data", () => {
    const wrapper = shallow(
      <CloudCoverage activity={cloudCoverageDataset.entries} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
