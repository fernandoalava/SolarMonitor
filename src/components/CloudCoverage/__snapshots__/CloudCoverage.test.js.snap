// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloudCoverage Renders correctly with data 1`] = `
<AreaChart
  barCategoryGap="10%"
  barGap={4}
  data={
    Array [
      Object {
        "sky_clarity": 98,
        "time": "18:00",
      },
      Object {
        "sky_clarity": 100,
        "time": "21:00",
      },
      Object {
        "sky_clarity": 100,
        "time": "00:00",
      },
      Object {
        "sky_clarity": 100,
        "time": "03:00",
      },
      Object {
        "sky_clarity": 100,
        "time": "06:00",
      },
      Object {
        "sky_clarity": 100,
        "time": "09:00",
      },
      Object {
        "sky_clarity": 100,
        "time": "12:00",
      },
      Object {
        "sky_clarity": 100,
        "time": "15:00",
      },
    ]
  }
  height={150}
  layout="horizontal"
  margin={
    Object {
      "bottom": 5,
      "left": 20,
      "right": 30,
      "top": 5,
    }
  }
  reverseStackOrder={false}
  stackOffset="none"
  width={500}
>
  <CartesianGrid
    fill="none"
    horizontal={true}
    horizontalFill={Array []}
    horizontalPoints={Array []}
    stroke="#ccc"
    strokeDasharray="3 3"
    vertical={true}
    verticalFill={Array []}
    verticalPoints={Array []}
  />
  <XAxis
    allowDataOverflow={false}
    allowDecimals={true}
    allowDuplicatedCategory={true}
    dataKey="time"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={30}
    hide={false}
    mirror={false}
    orientation="bottom"
    padding={
      Object {
        "left": 0,
        "right": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="category"
    width={0}
    xAxisId={0}
  />
  <YAxis
    allowDataOverflow={false}
    allowDecimals={true}
    allowDuplicatedCategory={true}
    dataKey="sky_clarity"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={0}
    hide={false}
    mirror={false}
    orientation="left"
    padding={
      Object {
        "bottom": 0,
        "top": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="number"
    width={60}
    yAxisId={0}
  />
  <Tooltip
    active={false}
    animationDuration={400}
    animationEasing="ease"
    contentStyle={Object {}}
    coordinate={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    cursor={true}
    cursorStyle={Object {}}
    filterNull={true}
    isAnimationActive={true}
    itemStyle={Object {}}
    labelStyle={Object {}}
    offset={10}
    separator=" : "
    useTranslate3d={false}
    viewBox={
      Object {
        "x1": 0,
        "x2": 0,
        "y1": 0,
        "y2": 0,
      }
    }
    wrapperStyle={Object {}}
  />
  <Legend
    align="center"
    iconSize={14}
    layout="horizontal"
    verticalAlign="bottom"
  />
  <Area
    activeDot={
      Object {
        "r": 8,
      }
    }
    animationBegin={0}
    animationDuration={1500}
    animationEasing="ease"
    connectNulls={false}
    dataKey="sky_clarity"
    dot={false}
    fill="#3182bd"
    fillOpacity={0.6}
    hide={false}
    isAnimationActive={true}
    legendType="line"
    name="% Sky Clarity"
    points={Array []}
    stroke="#8884d8"
    type="monotone"
    xAxisId={0}
    yAxisId={0}
  />
</AreaChart>
`;

exports[`CloudCoverage Renders correctly without data 1`] = `<WithStyles(ForwardRef(CircularProgress)) />`;
